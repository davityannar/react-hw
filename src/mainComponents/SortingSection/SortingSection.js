import React, { Component } from 'react'
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom'
import {asyncHandlingSubmitFormAction, changingSortByValue} from '../../actions/actionCreators'
import Filter from '../Search/SearchFilters/Filter/Filter'



import './SortingSection.scss'

export class SortingSection extends Component {
  constructor(props) {
    super(props)
  }
  render() {
    return (
      <div className='sorting-section'>
        <p className='movies-found'>
          {this.props.moviesFound} movies found
        </p>
        <div className='sort-filters'>
          <p>
            Sort by
          </p>
          
          <Filter inputId='release' onCustomChange={(e) => (this.props.handilngSortInputChange(e.target.value), this.props.changingSortBy(this.props.searchValue,this.props.searchBy, e.target.value))} inputClass='secondary-filter' inputName='sort' inputType='radio' inputValue='release_date' inputText='release date' checkedOption={this.props.sortBy === 'release_date' ? true : false} />
          <Filter inputId='rating' onCustomChange={(e) => (this.props.handilngSortInputChange(e.target.value), this.props.changingSortBy(this.props.searchValue, this.props.searchBy, e.target.value))} inputClass='secondary-filter' inputName='sort' inputType='radio' inputValue='vote_average' inputText='rating' checkedOption={this.props.sortBy === 'vote_average' ? true : false} />
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => ({searchValue: state.search.searchValue, searchBy: state.search.searchBy, sortBy: state.search.sortBy})
const mapDispatchToProps = (dispatch) => ({

  handilngSortInputChange(sortInputValue) {
    dispatch(changingSortByValue(sortInputValue))
  },
  changingSortBy(searchText, searchBy, sortBy) {
    dispatch(asyncHandlingSubmitFormAction(searchText, searchBy, sortBy))
  }
  
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(SortingSection))