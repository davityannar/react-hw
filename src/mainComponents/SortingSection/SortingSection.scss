.sorting-section {
    display: flex;
    justify-content: space-between;
    max-width: 1024px;
    margin: 20px auto 3%;
  }

  .movies-found {
    font-weight: bold;
  }

  .sort-filters {
    display: flex;
  
    p {
      margin: 0;
    }
  
    * {
      font-weight: bold;
    }
  }
  
  .secondary-filter {
    display: none;
    
    & + label {
      color: #000;
      transition: all .2s ease;
      margin-left: 15px;
      border-radius: 3px;
      cursor: pointer;
    }
    
    &:checked + label {
      color: #f55263;
      margin-left: 15px;
      transition: all .2s ease;
      border-radius: 3px;
      cursor: pointer;
    }
    }