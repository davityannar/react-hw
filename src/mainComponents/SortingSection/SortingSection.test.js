import React from 'react'
import SortingSection from './SortingSection'

describe('Tests SortingSection container', () => {
  it('Matching snapshot', () => {
    const container = shallow(<SortingSection />)

    expect(container).toMatchSnapshot();
  })
})