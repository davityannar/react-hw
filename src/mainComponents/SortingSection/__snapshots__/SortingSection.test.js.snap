// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests SortingSection container Matching snapshot 1`] = `
<div
  className="sorting-section"
>
  <p
    className="movies-found"
  >
    N movies found
  </p>
  <div
    className="sort-filters"
  >
    <p>
      Sort by
    </p>
    <input
      className="secondary-filter"
      defaultChecked={true}
      id="release"
      name="sort"
      type="radio"
      value="RELEASE"
    />
    <label
      htmlFor="release"
    >
      RELEASE
    </label>
    <input
      className="secondary-filter"
      id="rating"
      name="sort"
      type="radio"
      value="RATING"
    />
    <label
      htmlFor="rating"
    >
      RATING
    </label>
  </div>
</div>
`;
